File: backend/app/schemas/github.py
from pydantic import BaseModel, HttpUrl


class GitHubPRRequest(BaseModel):
    url: HttpUrl


class GitHubPRFile(BaseModel):
    filename: str
    status: str
    additions: int
    deletions: int
    changes: int
    patch: str | None = None


class GitHubPR(BaseModel):
    owner: str
    repo: str
    number: int
    title: str
    body: str | None = None
    state: str
    author: str
    html_url: str
    base_branch: str
    head_branch: str
    changed_files: int
    additions: int
    deletions: int
    files: list[GitHubPRFile]

File: frontend/src/types/github.ts
export interface GitHubPRFile {
  filename: string;
  status: string;
  additions: number;
  deletions: number;
  changes: number;
  patch: string | null;
}

export interface GitHubPRResponse {
  owner: string;
  repo: string;
  number: number;
  title: string;
  body: string | null;
  state: string;
  author: string;
  html_url: string;
  base_branch: string;
  head_branch: string;
  head_sha: string;
  changed_files: number;
  additions: number;
  deletions: number;
  files: GitHubPRFile[];
}
